aws_lambda_powertools/utilities/data_classes/sqs_event.py

```diff
@@ -174,6 +178,63 @@ def queue_url(self) -> str:
 
         return queue_url
 
+    @property
+    def decode_nested_s3_event(self) -> S3Event:
+        """Returns the nested `S3Event` object that is sent in the body of a SQS message.
+
+        Even though you can typecast the object returned by `record.json_body`
+        directly, this method is provided as a shortcut for convenience.
+
+        Notes
+        -----
+
+        This method does not validate whether the SQS message body is actually a valid S3 event.
+
+        Examples
+        --------
+
+        ```python
+        nested_event: S3Event = record.decode_nested_s3_event
+        ```
+        """
+        return self._decode_nested_event(S3Event)
+
+    @property
+    def decode_nested_sns_event(self) -> SNSMessage:
+        """Returns the nested `SNSMessage` object that is sent in the body of a SQS message.
+
+        Even though you can typecast the object returned by `record.json_body`
+        directly, this method is provided as a shortcut for convenience.
+
+        Notes
+        -----
+
+        This method does not validate whether the SQS message body is actually
+        a valid SNS message.
+
+        Examples
+        --------
+
+        ```python
+        nested_message: SNSMessage = record.decode_nested_sns_event
+        ```
+        """
+        return self._decode_nested_event(SNSMessage)
+
+    def _decode_nested_event(self, nested_event_class: Type[NestedEvent]) -> NestedEvent:
```

Comment:
I created this function because SQS events can encapsulate different types of events coming from SNS, S3, Kinesis, etc. I suggest that at some point we make it global, so that we can traverse multiple levels of nested events (e.g. S3 -> SNS -> SQS).

Reply:
@cowsandmilk hey David! I'm creating a RFC for multi-level nested events, so that we can discuss how the experience would look like from there.